### ISSUE TITLE
ui5-notification-li: new component

### ISSUE THREAD

**Issue by codefactor** [SAP/ui5-webcomponents#1478]
**Is your feature request related to a problem? Please describe.**
The SuccessFactors UI5 header supports a Notification List for displaying notification items when the "bell" icon is clicked. The Notification List which is part of sap.m library has no UI5 web component equivalent.

Here is a link to sample in sap.m:
http://veui5infra.dhcp.wdf.sap.corp:8080/sapui5-sdk-dist/#/entity/sap.m.NotificationListItem/sample/sap.m.sample.NotificationListItem

**Describe the solution you'd like**
At least the usage of NotificationList in SuccessFactors should be supported - but ideally the entire functionality of NotificationListItem/Group should be supported.

**Describe alternatives you've considered**
Using the custom list item and styling the contents manually.

**Additional context**

This is our UI5 version of the notification list (showing with sample data):
![image](https://user-images.githubusercontent.com/730310/79251706-5d2fc680-7e35-11ea-8415-58ff1a8112ae.png)

**Comment by MarcusNotheis** [comment]
Hi @codefactor,
in case you are using React in your application you might also look into the [Notification and NotificationGroup Components](https://sap.github.io/ui5-webcomponents-react/?path=/docs/3-components-notificationgroup--default-story) from the Web Component for React wrapper.
This might help you to bridge the time until the component is available as Web Component.

**Comment by codefactor** [comment]
@MarcusNotheis @ilhan007 @vladitasev ,

We are thinking to add this to a SuccessFactors web component version of the header which needs to be runnable from arbitrary page - it could be UI5 or React or legacy no-framework page. So using React in this case would be inconvenient. Though some of our pages do use React.

I wonder, is there any chance this might be started soon?

Also, I wonder, if this is a good or bad example of something that SuccessFactors team could consider to contribute to ui5-webcomponents as open source contributor? We are taking a look at this among other possibilities.

**Comment by vladitasev** [comment]
This will be prioritized for the next public release (RC8), but hopefully finished much sooner (before mid of May)

**Comment by vladitasev** [comment]
Note: the design has changed:
![image](https://user-images.githubusercontent.com/15844574/80488317-cef42e00-8966-11ea-8a04-afd486824d87.png)

We'll implement the latest Fiori design, as currently shown in the screenshot.

New component: `ui5-notification-li`:
 - slot for `avatar`
 - properties: `heading`, `icon`
 - `default slot` is the content - completely free
 - properties for close button and overflow button. TODO: determine the most convenient interface for providing overflow menu options. Could be similar to shellbar items for the different menus.
 - properties for the footer items, but not semantic such as "duration", etc... we just need to provide interface for the dots between them and styling. 
 - support for "Show more/show less"

**The new component is a list item.** Should be completely compatible with `ui5-list` in terms of design, selection modes (checkbox, radiobutton, active state, etc...)

**Important: notification list group is out of scope for the first release, but our implementation should allow for it in the future**

**Comment by vladitasev** [comment]
Update: scope has changed.

Notification list group will be part of the initial scope.

**Comment by codefactor** [comment]
@ilhan007 ,

It looks pretty good.
![image](https://user-images.githubusercontent.com/730310/82122076-328fa100-9746-11ea-8853-0b278c2198c2.png)

If you compare this with the original, there are only a few little differences:
1. The group header has less room because the button "See 1 old" is not on the next line
2. I couldn't find a place to put the text "Last: 4 months" in the group header -- which I think is the `sap.m.NotificationListGroup#datetime` property.

**Comment by codefactor** [comment]
@ilhan007 ,

One more thing - I wasn't able get it to work with a "busy-indicator" on any level outside of the parent level. To explain, when the user does something, such as mark an item as read or click on the "See 1 old" button, these activate asynchronous actions that may take some time. In OpenUI5 we could bind against the busy property, but in UI5 web components we can only wrap the thing we want with a <ui5-busyindicator> which doesn't work properly in this case. You can have a try by trying the following github:

```
git clone https://github.wdf.sap.corp/xweb/common-components
git co feature/header
cd common-components/packages/webcomponents
npm i
npm start
```
Then open http://localhost:8081/test-resources/pages/index.html?delay=3000

This will add artificial 3 second delay to the mocks and then open the notification side panel, and try out the "See 1 Old" button.

In the file:
https://github.wdf.sap.corp/xweb/common-components/blob/feature/header/packages/webcomponents/src/NotificationList.lit.js

You will find 2 places where the busy flag is passed down:
`renderNotificationGroup`
`renderNotificationItem`

In these cases they are given the busy, or parentBusy - where parentBusy is the busy flag on the group level and busy is the flag on the item level. So in my case those busy flags are not utilized right now -- you could try to add the <ui5-busyindicator> wrappers around the element and set the enabled flag on them based on these flags; however, you will find that the layout of the notification changes.

One other thing I might need some help -- for some reason any time I change the Javascript files, the build kicks in and sometimes the build has some fatal problems and refreshing the page no longer shows the header anymore. I am not sure what the issue is, maybe something wrong with the build tooling, or some difference with how I'm not using the "handlebars" for the view.

**Comment by ilhan007** [comment]
Hello @codefactor


1. **sap.m.NotificationListGroup "datetime" property**

It seems the NotificationListGroup "datetime" property is deprecated. 

https://openui5.hana.ondemand.com/api/sap.m.NotificationListGroup#controlProperties
<img width="901" alt="Screenshot 2020-05-17 at 9 57 51 AM" src="https://user-images.githubusercontent.com/15702139/82137934-fbdc8800-9824-11ea-9d3d-81cd6875e27b.png">

Also, I tried the nightly version, set "datetime", but it does not show up anywhere: http://veui5infra.dhcp.wdf.sap.corp:8080/snippix/#46822



2. **sap.m.NotificationListGroup actions**

I checked sap.m.NotificationListGroup and it behaves the same, neither the heading of the group item wraps, nor the actions. See the "Accept all" button does not wrap too.

<img width="306" alt="Screenshot 2020-05-17 at 10 10 33 AM" src="https://user-images.githubusercontent.com/15702139/82138117-b02ade00-9826-11ea-832b-46f81a60e3d8.png">


Best,
ilhan
